pep-9999.rst

```diff
@@ -0,0 +1,284 @@
+    ==================  ===========  =============================================
+
+
+A package index's credentials has the following attributes::
+
+    ==================  ===========  ===============================================
+    Attribute           Type         Description
+    ==================  ===========  ===============================================
+    cert                str          Path to certificate authority bundle (*.pem)
+    cert-client         str          Path to client certificate (*.pem)
+    enabled             bool         Whether the repository should be used or not (default: true)
+    package_index_name  str          The corresponding package index name, defined in ``config.toml``
+    password            str          The password for the repository
+    proxy_password      str          The password for the proxy server
+    proxy_username      str          The username for the proxy server
+    username            str          The username for the repository
+    ==================  ===========  ===============================================
+
+
+Tools preferring to look up credentials from other locations, such as keyring, are
+advised to use the full package index URL as the key, rather than the index name or
+hostname.
+
+
+The auth.toml file
+''''''''''''''''''
+
+Optionally, authentication can be defined in a separate ``auth.toml``. This way,
+credentials can be fully or partially separated from package index URLs.
+
+Example ``auth.toml``::
+
+    [[credentials]]
+    package_index_name = "myrepo1"
+    username = "foo"
+    password = "YmFyCg=="
+
+
+Package discovery
+-----------------
+
+PyPI.org
+''''''''
+
+The PyPI.org repository (``repositories.pypi``) does not have to be specified in the
+``config.toml``. If not specified, it will be used as the first package index by
+default. However, by explicitly defining it, the repository attributes can be used and
+it can be prioritized differently.
+
+
+Package index priority
+''''''''''''''''''''''
+
+Tools should respect priority by acting as if all lower priority package index were not
+specified when a higher priority package index satisfies the task. Tools that choose not
+to respect priority MUST display a warning to users when a non-zero priority value is
+being ignored. It is acceptable for tools to be less efficient in order to handle
+prioritized package indexes, for example, by not performing queries in parallel.
+
+
+Configuration file discovery
+----------------------------
+
+The locations of the configuration files (``config.toml`` and ``auth.toml``) vary slightly across platforms. You may have global, per-user and per-environment configuration:
+
+If multiple configuration files are found, then they are merged in the following order:
+
+1. The global files are read
+2. The per-user files are read
+3. The per-environment files are read
+
+
+Global
+''''''
+
+* Unix: ``/etc/.config/python/``
```

Comment:
This is non-standard. The correct path should be `/etc/python`.

Reply:
> I have left a few comments, but overall I like the idea.

👍 😃 

> One thought on wording: [PEP 508](https://www.python.org/dev/peps/pep-0503/) calls `https://pypi.org/simple` and the like a "Python package repository". I think distutils came up the with name "index" and it stuck. But to be consistent with other PEPs (including [PEP 458](https://www.python.org/dev/peps/pep-0458/) and [PEP 658](https://www.python.org/dev/peps/pep-0658/)) I would call them "repositories" or "Python package repositories".

Earlier, I actually referred to this as repositories:
https://github.com/fredrikaverpil/peps/pull/2/commits/d29137035b23ae448a57521ba0c8d3159150a008

But after having discussed some with @abn we settled on "indexes". Maybe we need to go back and look at this again because I think this discussion might have happened outside of this PR. I can't find it right now. I recall I went to the warehouse repo, the pypa repo and just read and realized "indexes" what was used everywhere. I recall reading that a repository basically holds these indexes (or indices).

> I also dislike the word "indexes" ;)

Hehe, yeah I am a bit torn here too...

Reply:
I updated this. Please have a look, and resolve this conversation if you are happy 😃